unlock-app/src/components/interface/MetadataTable.tsx

```diff
@@ -129,19 +129,16 @@ export const MetadataTable: React.FC<MetadataTableProps> = ({
     setCurrentLock(null)
   }
 
-  const isKeyValid = (metadata: KeyMetadata) => {
-    if (!metadata?.expiration) return false
-    if (metadata?.expiration.toLowerCase() === 'never') return true
-    const now = new Date().getTime()
-    const expiration = new Date(metadata?.expiration).getTime()
-    return expiration > now
+  const isKeyValid = (timestamp: KeyMetadata['expiration']) => {
+    const now = new Date().getTime() / 1000
+    return parseInt(timestamp) > now
```

Comment:
Unfortunately you need to handle the case where `timestamp` is MAX UINT (unlimited keys!) because parseInt will fail..

Reply:
should work also with `parseInt`, added check for `MAX_UINT` value